blockchain/chain.go

```diff
@@ -1665,6 +1665,73 @@ func (b *BlockChain) locateHeaders(locator BlockLocator, hashStop *chainhash.Has
 	return headers
 }
 
+// ReconsiderBlock takes a block hash and allows it to be revalidated.
+//
+// This function is safe for concurrent access.
+func (b *BlockChain) ReconsiderBlock(hash *chainhash.Hash) error {
+	b.chainLock.Lock()
+	defer b.chainLock.Unlock()
+	return b.reconsiderBlock(hash)
+}
+
+// reconsiderBlock takes a block hash and allows it to be revalidated.
+func (b *BlockChain) reconsiderBlock(hash *chainhash.Hash) error {
+	node := b.index.LookupNode(hash)
+	if node == nil {
+		err := fmt.Errorf("block %s is not known", hash)
+		return err
+	}
+
+	// No need to reconsider, it is already valid
+	if node.status.KnownValid() {
+		return nil
+	}
+
+	// Find fork point if one exists
+	forkNode := b.bestChain.FindFork(node)
+	if forkNode != nil {
+		// Roll back to the fork point and load all the nodes into a list
+		nodes := list.New()
+		for n := node; n != nil && n != forkNode; n = n.parent {
```

Comment:
I think this needs to be:
```
for n := node; n != nil && n != forkNode; n = n.parent {
	nodes.PushFront(n)
}

var next *list.Element
for e := nodes.Front(); e != nil; e = next {
	next = e.Next()
	n := nodes.Remove(e).(int)
}
```
https://play.golang.org/p/OqBgsd4rr-F

Think I screwed it up before.